pkg/provider/barbican/client.go

```diff
@@ -0,0 +1,243 @@
+	return err
+}
+
+// GetAllSecrets syncs multiple secrets from barbican provider into a single Kubernetes Secret.
+func (c *Client) GetAllSecrets(ctx context.Context, ref esv1beta1.ExternalSecretFind) (map[string][]byte, error) {
+	allPages, err := secrets.List(c.client, secrets.ListOpts{
+		Sort: "created:desc",
+	}).AllPages()
+	metrics.ObserveAPICall(constants.ProviderBarbican, constants.CallBarbicanGetAllSecrets, err)
+	if err != nil {
+		return nil, err
+	}
+
+	allSecrets, err := secrets.ExtractSecrets(allPages)
+	metrics.ObserveAPICall(constants.ProviderBarbican, constants.CallBarbicanGetAllSecrets, err)
+	if err != nil {
+		return nil, err
+	}
+
+	mapUUIDByUniqueNames := make(map[string]string, len(allSecrets))
+	for _, v := range allSecrets {
+		if _, ok := mapUUIDByUniqueNames[v.Name]; !ok {
+			mapUUIDByUniqueNames[v.Name] = extractIdFromRef(v.SecretRef)
+		}
+	}
+
+	result := make(map[string][]byte, len(mapUUIDByUniqueNames))
+
+	for name, id := range mapUUIDByUniqueNames {
+		payload, err := secrets.GetPayload(c.client, id, secrets.GetPayloadOpts{PayloadContentType: "*/*"}).Extract()
+		metrics.ObserveAPICall(constants.ProviderBarbican, constants.CallBarbicanGetAllSecrets, err)
+		if err != nil {
+			return nil, err
+		}
+
+		result[name] = payload
+	}
+
+	return result, nil
+}
+
+// GetSecret returns a single secret from the provider.
+func (c *Client) GetSecret(ctx context.Context, ref esv1beta1.ExternalSecretDataRemoteRef) ([]byte, error) {
+	var (
+		name   = ref.Key
+		uuid   = getUUIDFromRemoteRef(name)
+		secret *secrets.Secret
+		err    error
+	)
+
+	if uuid == "" {
+		secret, err = c.getByName(ctx, name)
+		if err != nil {
+			return nil, err
+		}
+		uuid = extractIdFromRef(secret.SecretRef)
+	}
+
+	return secrets.GetPayload(c.client, uuid, secrets.GetPayloadOpts{PayloadContentType: "*/*"}).Extract()
+}
+
+func (c *Client) Close(_ context.Context) error {
+	return nil
+}
+
+func (c *Client) Validate() (esv1beta1.ValidationResult, error) {
+	if c.storeKind == esv1beta1.ClusterSecretStoreKind && isReferentSpec(c.store) {
+		return esv1beta1.ValidationResultUnknown, nil
+	}
+	return esv1beta1.ValidationResultReady, nil
+}
+
+// GetSecretMap returns multiple k/v pairs from the provider.
+func (c *Client) GetSecretMap(ctx context.Context, ref esv1beta1.ExternalSecretDataRemoteRef) (map[string][]byte, error) {
+	data, err := c.GetSecret(ctx, ref)
+	metrics.ObserveAPICall(constants.ProviderBarbican, constants.CallBarbicanGetSecret, err)
+	if err != nil {
+		return nil, err
+	}
+
+	kv := make(map[string]json.RawMessage)
+	err = json.Unmarshal(data, &kv)
+	if err != nil {
+		return nil, fmt.Errorf(errJSONSecretUnmarshal, err)
+	}
+
+	secretData := make(map[string][]byte)
+	for k, v := range kv {
+		var strVal string
+		err = json.Unmarshal(v, &strVal)
+		if err == nil {
+			secretData[k] = []byte(strVal)
+		} else {
+			secretData[k] = v
+		}
+	}
```

Comment:
This implementation lacks the support for different JSON types (number/binary/bool/null). I think you can copy/paste from here:

https://github.com/external-secrets/external-secrets/blob/41cd1d36a4285be5b1d56a62428ea757d2a42f15/pkg/provider/vault/vault.go#L817-L830

I'll refactor that at some point in the future as this is copy/pasted across providers at this time.

Reply:
ok. I will copy/paste. Then we can unify it in one universal function